CMakeLists.txt

```
@@ -1,10 +1,26 @@
 cmake_minimum_required(VERSION 3.0.2)
-project( rauc-hawkbit-updater )
+project( rauc-hawkbit-updater VERSION 1.0 )
 
 option(WITH_SYSTEMD "Set to ON to create unit files and enable systemd startup"  OFF)
 option(BUILD_DOC    "Build documentation" OFF)
 option(QA_BUILD     "QA build (pedantic, with fatal errors)" OFF)
 
+# Set CPACK_SOURCE_IGNORE_FILES with same filter as .gitignore (to avoid redundancy)
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/.gitignore GITIGNORE_LINES)
+foreach (line ${GITIGNORE_LINES})
+    if(line MATCHES "^#")
+	    continue()
+    endif()
+    string(REPLACE "." "[.]" line "${line}")
+    string(REPLACE ";" "\\\\;" line "${line}")
+    string(REPLACE "*" ".*" line "${line}")
+    set(GITIGNORE_CONTENTS "${GITIGNORE_CONTENTS};/${line}")
+endforeach ()
+set(CPACK_SOURCE_IGNORE_FILES "/.git/;${GITIGNORE_CONTENTS}")
```

That's a pattern, so actual dots should be escaped:
```suggestion
set(CPACK_SOURCE_IGNORE_FILES "/\\.git/;${GITIGNORE_CONTENTS}")
```

Maybe also ignore the .gitignore here? Like this:
```suggestion
set(CPACK_SOURCE_IGNORE_FILES "/\\.git/;\\.gitignore;${GITIGNORE_CONTENTS}")
```

---

@Bastian-Krause  Good catch. I will leave the .gitignore in the dist package, if not cmake will break when trying to read the .gitignore file.

---

What about adding `if(EXISTS ..`? Is this too much hassle?

Also, could you replace the space indentation with tabs?

---

My bad, it is now fixed.

---

Thanks! Having this, we can add the `.gitignore` to `CPACK_SOURCE_IGNORE_FILES`, right?